library/core/src/iter/traits/accum.rs

```diff
@@ -43,28 +42,44 @@ macro_rules! integer_sum_product {
         #[$attr]
         impl Sum for $a {
             fn sum<I: Iterator<Item=Self>>(iter: I) -> Self {
-                iter.fold($zero, Add::add)
+                iter.fold(
+                    $zero,
+                    #[rustc_inherit_overflow_checks]
+                    |a, b| a + b,
```

Comment:
Hah, does this work? I half-expected it not to.

Reply:
Haha, this was [*your suggestion*](https://github.com/rust-lang/rust/issues/81721#issuecomment-772786852).. But yeah, it compiles. I think it works too, but I should check if it actually gets tested for this behaviour. :)

Reply:
Yeah, I was going off having seen `#[inline(never)}`, IIRC, on closures, but I wasn't sure it would work out of the box.